com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/VariablesRequestHandler.java

```diff
@@ -230,14 +230,21 @@
             }
 
             int referenceId = 0;
+
+            boolean containerIsArray = containerNode.getProxiedVariable() instanceof ArrayReference;
+            String evaluateName;
+
             if (indexedVariables > 0 || (indexedVariables < 0 && VariableUtils.hasChildren(value, showStaticVariables))) {
-                VariableProxy varProxy = new VariableProxy(containerNode.getThread(), containerNode.getScope(), value);
+                VariableProxy varProxy = new VariableProxy(containerNode.getThread(), containerNode.getScope(), value, containerNode, javaVariable.name);
                 referenceId = context.getRecyclableIdPool().addObject(containerNode.getThreadId(), varProxy);
+                evaluateName = varProxy.getEvaluateName();
```

Comment:
For  the first solution, when the hiearchy is deep, the evaluateName is already complex, in my opinion this isn't an issue. 

I really dont like the second solution.

So, if you agree, I'll implement the solution 1.

Reply:
In solution 1, it may be a little tricky for generating the correct evaluate expression for `Collection` type. For example, in order to copy the **n**th element's children, the evaluate expression is like `collection.toArray()[n].x`, this will report compilation error because of java type check.